circuit_knitting_toolbox/circuit_cutting/qpd/qpd.py

```diff
@@ -384,23 +384,24 @@ def _decompose_qpd_measurements(
         if instruction.operation.name.lower() == "qpd_measure"
     ]
 
-    # Create a classical register for the qpd measurement results.  This is
-    # partly for convenience, partly to work around
-    # https://github.com/Qiskit/qiskit-aer/issues/1660.
-    reg = ClassicalRegister(len(qpd_measure_ids), name="qpd_measurements")
-    circuit.add_register(reg)
-
-    # Place the measurement instructions
-    for idx, i in enumerate(qpd_measure_ids):
-        gate = circuit.data[i]
-        inst = CircuitInstruction(
-            operation=Measure(), qubits=[gate.qubits], clbits=[reg[idx]]
-        )
-        circuit.data[i] = inst
+    if len(qpd_measure_ids) > 0:
```

Comment:
I snuck this change in, but I want to draw attention to it.

Let's not add empty QPD registers if the circuit doesn't have any mid-circuit measurements. This came up when I was writing tests for the cutting code. I think it's more intuitive for a circuit to not have a qpd register if no qpd measurements are expected

Reply:
OK, yeah, I noticed this and I am glad you brought this up.  I am not sure I agree -- I think that having the register be there consistently is more in line with expectation.  I also don't think that this allowed the tests to be simplified all that much (I think you mentioned earlier that this was part of the motivation for doing this?)